walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,87 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors; 
+import java.util.stream.Stream; 
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    int dayStart = TimeRange.START_OF_DAY;
+    int dayEnd = TimeRange.END_OF_DAY;
+    int start = dayStart;
+    int end;
+    TimeRange previousTime = TimeRange.fromStartDuration(0,0);
+    
+    List<TimeRange> sortedTimeRanges = getAllAttendeeRanges(events, request);
+    ArrayList<TimeRange> possibleTimeRanges = new ArrayList<>();
```

Comment:
Same with all of these -- what do the time ranges represent?

Reply:
So the time ranges represent the free (non-scheduled) time between events they're scheduled for. To clarify a bit if an individual has an event from (1:30 - 2:00) and another event between (3:00 - 4:30) the time range would be (2:00 - 3:00) the range of non-scheduled time between their scheduled events.

Reply:
So these time ranges represent all the times that individuals that are requested to attend the meeting have on their calendar, I use these to retrieve start and end times to find availability for everyone in the meeting request list.

Reply:
I've changed updated the names to help reflect their purpose a bit more.